draco/core/io.py

```diff
@@ -75,40 +76,98 @@
 
 
 def _list_of_filelists(files):
-    # Take in a list of lists/glob patterns of filenames
+    """
+    Take in a list of lists/glob patterns of filenames
+
+    Parameters
+    ----------
+    files : List[str] or List[List[str]]
+        A path or glob pattern (e.g. /my/data/*.h5) or a list of those (or a list of lists of those).
+
+    Raises
+    ------
+    ConfigError
+        If files has the wrong format or refers to a file that doesn't exist.
+
+    Returns
+    -------
+    List[List[str]]
+        The input file list list. Any glob patterns will be flattened to file path string lists.
+    """
     import glob
 
     f2 = []
 
     for filelist in files:
 
         if isinstance(filelist, str):
+            if "*" not in filelist and not os.path.isfile(filelist):
+                raise ConfigError("File not found: %s" % filelist)
             filelist = glob.glob(filelist)
         elif isinstance(filelist, list):
-            pass
+            for i in range(len(filelist)):
+                filelist[i] = _list_or_glob(filelist[i])
         else:
-            raise Exception("Must be list or glob pattern.")
-        f2.append(filelist)
+            raise ConfigError("Must be list or glob pattern.")
+        f2 = f2 + filelist
 
     return f2
 
 
 def _list_or_glob(files):
-    # Take in a list of lists/glob patterns of filenames
+    """
+    Take in a list of lists/glob patterns of filenames
+
+    Parameters
+    ----------
+    files : str or List[str]
+        A path or glob pattern (e.g. /my/data/*.h5) or a list of those
+
+    Returns
+    -------
+    List[str]
+        The input file list. Any glob patterns will be flattened to file path string lists.
+
+    Raises
+    ------
+    ConfigError
+        If files has the wrong type or if it refers to a file that doesn't exist.
+    """
     import glob
 
     if isinstance(files, str):
+        if "*" not in files and not os.path.isfile(files):
+            raise ConfigError("File not found: %s" % files)
         files = sorted(glob.glob(files))
     elif isinstance(files, list):
-        pass
+        parsed_files = []
+        for f in files:
+            parsed_files = parsed_files + _list_or_glob(f)
+        files = parsed_files
     else:
-        raise ValueError("Argument must be list or glob pattern, got %s" % repr(files))
-
+        raise ConfigError("Argument must be list or glob pattern, got %s" % repr(files))
     return files
 
 
 def _list_of_filegroups(groups):
-    # Process a file group/groups
+    """
+    Process a file group/groups
+
+    Parameters
+    ----------
+    groups : List[Dict] or Dict
+        Dicts should contain keys 'files': An iterable with file path or glob pattern strings, 'tag': the group tag str
+
+    Returns
+    -------
+    List[Dict]
```

Comment:
If you're going to give the types in this format, you might as well make them actual type annotations, and drop them from the docstring. Sphinx should be able to pull them out as expected.